language/llama2-70b/dataset.py

```diff
@@ -61,21 +63,36 @@ def load_processed_dataset(self):
             self.input_ids.append(input_ids)
             self.attention_masks.append(attn_mask)
             self.input_lens.append(input_ids.shape[-1])
+        print("Finished loading dataset.")
 
 
     def postProcess(self, out_tokens, input_seq_lens=None, query_id_list=None, sample_index_list=None):
         """ Postprocesses output prediction """
 
         #TODO: Create response object in postProcess(?)
+        """
         preds = []
         for i in range(out_tokens.shape[0]):
             #pred = out_tokens[i].reshape(-1).cpu().numpy() # Slice up to original input length as below?
 
             input_len = input_seq_lens[i] if input_seq_lens else 0
             pred = out_tokens[i, input_len:].reshape(-1).cpu().numpy()
             preds.append(pred)
-        
-        return preds
+        """
+        # Everything is padded to max_len (1024), so prune the input and parse to numpy
+        output_seq = out_tokens[:, 1024:].cpu().numpy()
```

Comment:
Note that the script generates outputs with duplicated EOS tokens. It causes the issue described in https://github.com/mlcommons/inference/issues/1545 .

I'm not sure should script or loadgen take care of removing all tokens after the first EOS.

Reply:
I think the loadgen should take care of the EOS removal to ensure fair accuracy and perf measurement. SUT can choose to remove it or not based on the implementation.

Reply:
I agree with Michal. In general LoadGen should be agnostic to the output of the model and not do any kind of post-processing. I think this should include dropping parts of the output

I also don't see the issue in removing this in the SUT, this should be negligible in comparison to the time it takes to query the model, shouldn't it?